nautilus_core/indicators/src/average/hma.rs

```diff
@@ -0,0 +1,246 @@
+        stubs::*,
+    };
+
+    #[rstest]
+    fn test_hma_initialized(indicator_hma_10: HullMovingAverage) {
+        let display_str = format!("{indicator_hma_10}");
+        assert_eq!(display_str, "HullMovingAverage(10)");
+        assert_eq!(indicator_hma_10.period, 10);
+        assert!(!indicator_hma_10.is_initialized);
+        assert!(!indicator_hma_10.has_inputs);
+    }
+
+    #[rstest]
+    fn test_initialized_with_required_input(mut indicator_hma_10: HullMovingAverage) {
+        for i in 1..10 {
+            indicator_hma_10.update_raw(f64::from(i));
+        }
+        assert!(!indicator_hma_10.is_initialized);
+        indicator_hma_10.update_raw(10.0);
+        assert!(indicator_hma_10.is_initialized);
+    }
+
+    #[rstest]
+    fn test_value_with_one_input(mut indicator_hma_10: HullMovingAverage) {
+        indicator_hma_10.update_raw(1.0);
+        assert_eq!(indicator_hma_10.value, 1.0);
+    }
+
+    #[rstest]
+    fn test_value_with_three_inputs(mut indicator_hma_10: HullMovingAverage) {
+        indicator_hma_10.update_raw(1.0);
+        indicator_hma_10.update_raw(2.0);
+        indicator_hma_10.update_raw(3.0);
+        assert_eq!(indicator_hma_10.value, 1.824_561_403_508_772);
+    }
+
+    #[rstest]
+    fn test_value_with_ten_inputs(mut indicator_hma_10: HullMovingAverage) {
+        indicator_hma_10.update_raw(1.00000);
+        indicator_hma_10.update_raw(1.00010);
+        indicator_hma_10.update_raw(1.00020);
+        indicator_hma_10.update_raw(1.00030);
+        indicator_hma_10.update_raw(1.00040);
+        indicator_hma_10.update_raw(1.00050);
+        indicator_hma_10.update_raw(1.00040);
+        indicator_hma_10.update_raw(1.00030);
+        indicator_hma_10.update_raw(1.00020);
+        indicator_hma_10.update_raw(1.00010);
+        indicator_hma_10.update_raw(1.00000);
+        assert_eq!(indicator_hma_10.value, 1.000_140_392_817_059_8);
+    }
+
+    #[rstest]
+    fn test_handle_quote_tick(mut indicator_hma_10: HullMovingAverage, quote_tick: QuoteTick) {
+        indicator_hma_10.handle_quote_tick(&quote_tick);
+        assert_eq!(indicator_hma_10.value, 1501.0);
+    }
+
+    #[rstest]
+    fn test_handle_trade_tick(mut indicator_hma_10: HullMovingAverage, trade_tick: TradeTick) {
+        indicator_hma_10.handle_trade_tick(&trade_tick);
+        assert_eq!(indicator_hma_10.value, 1500.0);
+    }
+
+    #[rstest]
+    fn test_handle_bar(
+        mut indicator_hma_10: HullMovingAverage,
+        bar_ethusdt_binance_minute_bid: Bar,
+    ) {
+        indicator_hma_10.handle_bar(&bar_ethusdt_binance_minute_bid);
+        assert_eq!(indicator_hma_10.value, 1522.0);
+        assert!(indicator_hma_10.has_inputs);
+        assert!(!indicator_hma_10.is_initialized);
+    }
+
+    #[rstest]
+    fn test_reset(mut indicator_hma_10: HullMovingAverage) {
+        indicator_hma_10.update_raw(1.0);
+        assert_eq!(indicator_hma_10.count, 1);
+        indicator_hma_10.reset();
```

Comment:
You could better test this, running few `update_raw` then `reset` and few again `update_raw` would fail, because three ma's didn't change state in reset method, and this test would fail